core/frontend/src/extension/Extension.ts

```diff
@@ -0,0 +1,58 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Extensions
+ */
+
+/**
+ * @beta
+ */
+export enum ActivationEvent {
+  onStartup = "onStartup",
+}
+
+/**
+ * @beta
+ */
+export type ResolveFunc = () => Promise<any>;
+
+/** Defines the format of an Extension manifest
+ * @beta
+*/
+export interface ExtensionManifest {
+  /** The extension name */
```

Comment:
Any uniqueness constraints? Difference from displayName?

Reply:
no constraints at this time. We aren't requiring uniqueness for build-time. That said, I think we may want to do so in the future (thinking mostly of runtime extensions). @calebmshafer @aruniverse  thoughts?

Reply:
Based name and displayName off of what vscode does, see https://code.visualstudio.com/api/references/extension-manifest

for name i was thinking itd end up being the name defined in the pkg.json, and the displayName, if provided, would be the name shown when we eventually have an extension store/ui

i dont think the name or displayName need to be unique, we should have some guid on our end